mkdocs_static_i18n/reconfigure.py

```diff
@@ -0,0 +1,370 @@
+    if "locale" in theme._vars:
+        theme._vars["locale"] = localization.parse_locale(locale)
+    return theme
+
+
+def reconfigure_material_theme(
+    locale: str, i18n_config: MkDocsConfig, plugin_config: dict, build_languages
+) -> MkDocsConfig:
+    # set theme language
+    if "language" in i18n_config.theme._vars:
+        i18n_config.theme._vars["language"] = locale
+    # site language selector reconfiguration can be disabled
+    if plugin_config["material_alternate"] is True:
+        # configure extra.alternate language switcher
+        if len(build_languages) > 1:
+            # user has setup its own extra.alternate
+            # warn him if it's poorly configured
+            if "alternate" in i18n_config["extra"]:
+                for alternate in i18n_config["extra"]["alternate"]:
+                    if not alternate.get("link", "").startswith("./"):
+                        log.info(
+                            "The 'extra.alternate' configuration contains a "
+                            "'link' option that should starts with './' in "
+                            f"{alternate}"
+                        )
+            # configure the extra.alternate for the user
+            else:
+                i18n_config["extra"]["alternate"] = []
+                # Add index.html file name when used with
+                # use_directory_urls = True
+                link_suffix = ""
+                if i18n_config.get("use_directory_urls") is False:
+                    link_suffix = "index.html"
+                # setup language switcher
+                for language in build_languages:
+                    lang_config = plugin_config["languages"][language]
+                    # skip language if not built
+                    if lang_config["build"] is False:
+                        continue
+                    i18n_config["extra"]["alternate"].append(
+                        {
+                            "name": f"{lang_config['name']}",
+                            "link": f"{lang_config['link']}{link_suffix}",
+                            "fixed_link": lang_config["fixed_link"],
+                            "lang": language,
+                        }
+                    )
+    return i18n_config
+
+
+def reconfigure_search_plugin(
+    i18n_config: MkDocsConfig,
+    plugin_config: dict,
+    build_languages,
+    search_plugin_name: str,
+    search_plugin,
+):
+    # search plugin reconfiguration can be disabled
+    if plugin_config["search_reconfigure"]:
+        search_langs = search_plugin.config["lang"] or []
+        for language in build_languages:
+            if language in LUNR_LANGUAGES:
+                if language not in search_langs:
+                    search_langs.append(language)
+                    log.info(
+                        f"Adding '{language}' to the '{search_plugin_name}' plugin 'lang' option"
+                    )
+            else:
+                log.warning(
```

Comment:
This should not be a warning, that would break the `strict` run, since the user can't do much about it.